physical/consul.go

```diff
@@ -56,19 +58,14 @@ type ConsulBackend struct {
 	kv                  *api.KV
 	permitPool          *PermitPool
 	serviceLock         sync.RWMutex
-	service             *api.AgentServiceRegistration
-	sealedCheck         *api.AgentCheckRegistration
-	registrationLock    int64
 	advertiseHost       string
 	advertisePort       int64
-	consulClientConf    *api.Config
 	serviceName         string
-	running             bool
-	active              bool
-	unsealed            bool
 	disableRegistration bool
 	checkTimeout        time.Duration
-	checkTimer          *time.Timer
+
+	notifyActiveCh chan notifyEvent
```

Comment:
Where is the channel actually allocated? It should be created with buffer size 1 in newConsulBackend